services/api/db/serializers.py

```diff
@@ -0,0 +1,101 @@
+        try:
+            validate_email(email)
+        except:
+            raise serializers.ValidationError("Please provide a valid email address.")
+        if validate_confirm_password(password,confirm_password):
+            return data
+    def create(self, validated_data):
+        return User.objects.create_user(**validated_data)
+    
+class SignInEndPointSerializer(serializers.ModelSerializer):
+    email = serializers.EmailField(
+        max_length=255,
+        style={'input_type': 'email', 'placeholder': 'Email Address'}
+    )
+    class Meta:
+        model = User
+        fields = ['email', 'password']
+
+class UserProfileSerialiser(serializers.ModelSerializer):
+    class Meta:
+        model = User
+        fields = ['id', 'email', 'name']
+
+class UserChangePasswordSerialiser(serializers.Serializer):
+    #New password type could be customised here.
+    password = serializers.CharField(max_length=255, style={'input_type' : 'password'}, write_only=True)
+    confirm_password = serializers.CharField(max_length=255, style={'input_type' : 'password'}, write_only=True)
+    class Meta:
+        fields = ['password', 'confirm_password']
+    def validate(self, attrs):
+        password = attrs.get('password')
+        confirm_password = attrs.get('confirm_password')
+        if validate_confirm_password(password,confirm_password):
+            return attrs
+
+    
+class SendPasswordResetEmailSerializer(serializers.Serializer):
+    email = serializers.EmailField(
+        max_length=255,
+        style={'input_type': 'email', 'placeholder': 'Email Address'}
+    )
+    class Meta:
+        fields = ['email']
+    def validate(self, attrs):
+        email = attrs.get('email')
+        if User.objects.filter(email=email).exists():
+            user = User.objects.get(email = email)
+            uid = urlsafe_base64_encode(force_bytes(user.id))
+            token = PasswordResetTokenGenerator().make_token(user)
+            """
+            Instead of local server, desired domain server could be added here.
+            """
+            link = f'http://localhost:3000/api/user/reset/{uid}/{token}'
+            return attrs
+        else:
+            raise serializers.ValidationError('You are not a Registered User')
+    
+class UserPasswordResetSerializer(serializers.Serializer):
+    password = serializers.CharField(max_length=255, style={'input_type':'password'}, write_only=True)
+    confirm_password = serializers.CharField(max_length=255, style={'input_type':'password'}, write_only=True)
+    class Meta:
+        fields = ['password', 'confirm_password']
+
+    def validate(self, attrs):
+        password = attrs.get('password')
+        confirm_password = attrs.get('confirm_password')
+        uid = self.context.get('uid')
+        token = self.context.get('token')
+        validate_confirm_password(password,confirm_password)
+        id = smart_str(urlsafe_base64_decode(uid))
+        user = User.objects.get(id=id)
```

Comment:
This could potentially throw runtime error, which should be handled.

Reply:
handled it